addons/binding/org.openhab.binding.coolmasternet/src/main/java/org/openhab/binding/coolmasternet/handler/HVACHandler.java

```diff
@@ -0,0 +1,141 @@
+                    controller.sendCommand(String.format("%s %s", command, uid));
+                } else if (channel.endsWith(FAN) && command instanceof StringType) {
+                    controller.sendCommand(String.format("fspeed %s %s", uid, command));
+                } else if (channel.endsWith(LOUVRE) && command instanceof StringType) {
+                    controller.sendCommand(String.format("swing %s %s", uid, command));
+                }
+            }
+        } catch (CoolMasterClientError e) {
+            logger.error("Failed to set channel {} -> {}: {}", channel, command, e.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initialising CoolMasterNet HVAC handler...");
+        super.initialize();
+        controller = (ControllerHandler) getBridge().getHandler();
+    }
+
+    /* Update this HVAC unit's properties from the controller */
+    public void refresh() {
+        ThingUID thinguid = getThing().getUID();
+
+        OnOffType on = "1".equals(query("o")) ? OnOffType.ON : OnOffType.OFF;
+        updateState(new ChannelUID(thinguid, ON), on);
+        updateState(new ChannelUID(thinguid, CURRENT_TEMP), new DecimalType(query("a")));
+        updateState(new ChannelUID(thinguid, SET_TEMP), new DecimalType(query("t")));
+        String mode = modeNumToStr.getOrDefault(query("m"), null);
+        updateState(new ChannelUID(thinguid, MODE), new StringType(mode));
+        updateState(new ChannelUID(thinguid, LOUVRE), new StringType(query("s")));
+        String fan = fanNumToStr.getOrDefault(query("f"), null);
+        updateState(new ChannelUID(thinguid, FAN), new StringType(fan));
+    }
+
+    private String query(String query_char) {
+        String cmn_uid = (String) getConfig().get(UID);
+        String command = String.format("query %s %s", cmn_uid, query_char);
+        try {
+            return controller.sendCommand(command);
+        } catch (CoolMasterClientError e) {
+            logger.error("Query {} failed: {}", command, e.getMessage());
+            return null; /* passing back null sets an invalid value on the channel */
+        }
+    }
+
+    /*
+     * The coolmasternet query command returns numbers 0-5 for operation modes,
+     * but these don't map to any mode you can set on the device, so we use this
+     * lookup table.
+     */
+    private static final Map<String, String> modeNumToStr;
+    static {
+        modeNumToStr = new HashMap<>();
+        modeNumToStr.put("0", "cool");
+        modeNumToStr.put("1", "heat");
+        modeNumToStr.put("2", "auto");
+        modeNumToStr.put("3", "dry");
+        /* 4=='haux' but this mode doesn't have an equivalent command to set it! */
+        modeNumToStr.put("4", "heat");
+        modeNumToStr.put("5", "fan");
+    }
+
+    /*
+     * The coolmasternet query command returns numbers 0-5 for fan speed,
+     * but the fan command uses single-letter abbreviations. Yay consistency?
```

Comment:
I guess it's one way or the other - all the other binding commands use the same representation as the protocol commands. Switching the fan commands to be numbers removes the need for this lookup table on query results, but still need a lookup table when sending commands. I think this is slightly more consistent with the way the other commands are implemented.

(Will remove my snarky 'Yay Consistency?', though. :).)